packages/core/src/auth/sso/server.ts

```diff
@@ -190,18 +191,20 @@ export class AuthSSOServer {
         res.end(error.message)
 
         // Send the response back to the editor
-        this.deferred?.reject(error)
+        this.deferred?.resolve(Result.err(error))
     }
 
-    public waitForAuthorization(): Promise<string> {
+    public waitForAuthorization(): Promise<Result<string>> {
         return Promise.race([
             this.authenticationPromise,
-            new Promise<string>((_, reject) => {
+            new Promise<Result<string>>((_, reject) => {
                 globals.clock.setTimeout(() => {
                     reject(
-                        new ToolkitError('Timed-out waiting for browser login flow to complete', {
-                            code: 'TimedOut',
-                        })
+                        Result.err(
```

Comment:
should it be `resolve()` now? or maybe `reject(new ToolkitError(...))` still "works" for this case?